### ISSUE TITLE
Need a way to link against libcore

### ISSUE THREAD

**Issue by mcoffin** [jensnockert/dueboot#7]
There needs to be a decent way to link against libcore if we want to be able to write idiomatic Rust with this.

I have put (admittedly hacked) together a preliminary version in my master branch (currently mcoffin@58bde74860886e9a39e12df094b2ab5147059f2f). Unfortunately, I stripped the whole build system in the process because I got frustrated. So, before we even THINK about merging it, we'll have to either update the new build system or convert the hack I used back over to the old one.

Basically, I build libcore for `arm-unkown-linux-gnueabi`, then rust will grab its metadata from `libcore.rlib` and then I recompile libcore, this time emitting llvm-ir, then compile and manually link the result with the main crate.

Thoughts?

**Comment by mcoffin** [comment]
It might almost be easier to come up with a way to just compile rustc for a freestanding target. I know that was in the works but I'm not sure where they are with it right now.

**Comment by polarathene** [comment]
@mcoffin I have a Due and I'm looking into if I can port my project to use Rust. It's handling packet data through RS485 via a shield, parsing that and sending it out to my PC over usb-serial(wifi at a later stage hopefully).

I see your fork had a little bit of work since this issue but has since ceased. Is this project or your fork still the way to go, or would you advise other projects that are more active like [Zinc](https://github.com/hackndev/zinc)? Any other links/resources you could point me towards would be appreciated.

I'm not quite sure where to ask questions such as if I will lose Arduino methods and what difficulty there might be to re-implement(though I've read of access to externs) as well as some rust limitations(no std lib)? Is there a community established? I take it these projects are too specific/niche right now for Arduino and Rust communities for discussion.

**Comment by mcoffin** [comment]
@polarathene, Zinc is more of an exploration in to writing a **full** embedded stack in rust. If you're concerned about keeping access to arduino methods, and not with writing register-level code in Rust, then I would do something more along the lines of this project. I'll write a little more extensive post later.

**Comment by polarathene** [comment]
@mcoffin thanks for the response :) zinc might be interesting at a later stage, as I'm relatively new to embedded and will have enough trouble getting over the initial learning hurdle with Rust I'd like to keep it simple as possible.

If you're going to contribute an extensive post, it may have a better audience on [my /r/rust thread](https://www.reddit.com/r/rust/comments/4rhmqx/rust_and_armmbed_arduino_and_embedded_rust_in/) I created. That should increase it's visibility/helpfulness :) Thanks in advance! shepmaster/Jake Goulding has [mentioned putting together a `libarduino.rs`](https://www.reddit.com/r/rust/comments/4oxfe8/will_we_ever_see_rust_for_arduino/d4gjwwc) he is working on AVR/AVR-LLVM support but I'd hope that lib could be used with Cortex-M MCUs too.

**Comment by mcoffin** [comment]
If anybody is still interested in this, the way to go is to use [`xargo`](https://crates.io/crates/xargo) for the build.
